### PR TITLE
wip for making generate a bit more oo

### PR DESCRIPTION
**Pull request by ellduin**
fewer dict strings and more properties

everything tested with:
python3 generate.py && diff -ur out out_orig 

so it really does generate the same output as before

### CODE REVIEW

File: generate.py [hunk 1]
```diff
@@ -298,225 +298,274 @@ def calculate_sha1(filepath):
...
+        self.index_category = metadata['index_category']
+        self.category = metadata['category'] # unused?
+        self.target_filename = metadata['target_filename']
+        self.downloads = metadata['downloads']
+        self.description = metadata['description']
+        self.src_img_urls = metadata['src_img_urls']
+        self.html = metadata['html']
+        self.txt = metadata['txt']
+        self.img_urls = metadata['img_urls']
+        self.path = metadata['path'] # unused?
```
Comment (mist64) [inline generate.py:326]: mark with TODO or XXX or something

File: generate.py [hunk 2]
```diff
@@ -298,225 +298,274 @@ def calculate_sha1(filepath):
...
+          self.issue_key = issue_key
+
+          # used later on
+          self.toc_order = toc_order
+          self.pubdate = pubdate
+          self.pdf_filename = pdf_filename
+          self.issue_dir_name = issue_dir_name
+          self.listings = listings
+
+      else:
```
Comment (mist64) [inline generate.py:382]: swap then and else

File: generate.py [hunk 3]
```diff
@@ -551,54 +600,57 @@ def all_type_in_articles_grouped_by_index_category(self):
...
+    pubdate = issue.pubdate
     # Add index as half-days to the date
-    pubdate += timedelta(hours=HOURS_PER_ARTICLE * article['index'])
+    pubdate += timedelta(hours=HOURS_PER_ARTICLE * article_dict['index'])
     return pubdate
 
 
+def optional_issue_prefix(path, issue, prepend_issue_dir=False):
+    if prepend_issue_dir:
+        issue_dir = issue.issue_dir_name
```
Comment (mist64) [inline generate.py:644]: combine lines

**Review by mist64 — APPROVED** [review]

**Comment by ellduin** [comment]
comments will be fixed in cleanup part 2

**Pull request merged**